### DISCUSSION TITLE
Building Firebird 1.5.6 from source on x64 system

### DISCUSSION THREAD

**Discussion by Morc001** [FirebirdSQL/firebird#7432] [General]
Hi Anybody,

  I tried to build Firebird 1.5.6 from source on quite a few OS versions, but all failed at some point. Tried Ubuntu version from 18.04 to 8.04. Newer releases fail at running autogen.sh and generating the Makefile/configure script due to missing m4 macros. 10.04 I believe allowed me to generate a configure script and the configure script ran through, but then make fails with alloc.h defining the new method as static. I tried Fedora version 14 through 8, all fail the same way as Ubuntu does (m4 or C/C++ code). What OS were the 1.5.6 binaries built originally? I thought RedHat/Fedora because RPMs are available. Can anyone help me? Or alternatively can applications/databases be used with 2.5 without any modifications?

  Thanks,

  Morc

**Comment by AlexPeshkoff** [comment]
On 12/21/22 20:15, Morc001 wrote:
>
> Hi Anybody,
>
> I tried to build Firebird 1.5.6 from source on quite a few OS 
> versions, but all failed at some point. Tried Ubuntu version from 
> 18.04 to 8.04. Newer releases fail at running autogen.sh and 
> generating the Makefile/configure script due to missing m4 macros. 
> 10.04 I believe allowed me to generate a configure script and the 
> configure script ran through, but then make fails with alloc.h 
> defining the new method as static. I tried Fedora version 14 through 
> 8, all fail the same way as Ubuntu does (m4 or C/C++ code). What OS 
> were the 1.5.6 binaries built originally? I thought RedHat/Fedora 
> because RPMs are available. Can anyone help me?
>

FB1.5 does not support x64, only x86 (32 bit). It's VERY old and not 
supported for many years.
Possible solution is to use 1.5 CS on 32-bit system - due to it's 
multiprocess nature it breaks 4Gb RAM limit efficiently.

> Or alternatively can applications/databases be used with 2.5 without 
> any modifications?
>

Almost yes, but there may be minor issues. For example with reserved SQL 
words which might be used in 1.5 as normal names for columns and 
variables, like RETURN.

PS. Be aware that even 2.5 is not supported by community any more. It 
was released >10 years ago.

**Reply by Morc001** [reply]
Hi Alex,

  Thanks for the quick reply! I understood that 1.5 is veeeery old and not supported anymore. But I need to run multiple Firebird versions alongside each other on not-outdated systems. Ubuntu stopped supporting 32 bit OS with 18.04, which is ending its lifetime in April. 20.04 has some 32 bit libraries, but not all that are needed to run Firebird 1.5. Hence me trying to build a 64 bit version. I found one forum post (https://forums.raspberrypi.com/viewtopic.php?t=185884) where someone tried and then managed to build this version for Raspberry Pi3, although it's not clear whether it was a 32 or 64 bit system.

  Anyhow by saying "FB1.5 does not support x64" you mean you have some low-level code that breaks on x64, or that it was never tested on x64 and in theory it could work?

  And more importantly, do you know what OS/release the currently available RPM and tar.gz builds were made on?

**Comment by AlexPeshkoff** [comment]
On 12/21/22 20:52, Morc001 wrote:
>
> Hi Alex,
>

> Anyhow by saying "FB1.5 does not support x64" you mean you have some 
> low-level code that breaks on x64, or that it was never tested on x64 
> and in theory it could work?
>

Tried build it in 2005, build failed (do not remember how), soon after 
was decided not to waste resources on 1.5 x64 port but concentrate on 2.0.
If you succeed to build it on x64 remember - databases will need 
backup/restore to change ODS, it will defintely not match with x86! 
(This was fixed in 2.0)

> And more importantly, do you know what OS/release the currently 
> available RPM and tar.gz builds were made on?
>

RH8 (not current RHEL but one released around year 2002 or 2003 - distro 
which was renamed to Fedora).
Super server with NPTL support - RH9 with gcc upgraded to 3.2.x, upgrade 
was needed due to exceptions handling MT issue.

**Reply by Morc001** [reply]
Thanks for the replies and insider info :) We'll see if we can get things going with FB2.5.

Fedora Core 1 (yarrow) builds the binaries, but fails to create the empty.fdb with "Implementation limit exceeded", so I gave up for the moment.
